### DISCUSSION TITLE
Algorithmic Base Pay Recommendations

### DISCUSSION THREAD

**Discussion by 0x4007** [ubiquity/ubiquibot#487] [Ideas]
It would be interesting if the UbiquiBot makes recommendations for project maintainers on what a contributor's base pay should be, based on different recommendation strategies. This is not intended to automatically apply the changes, it's just for project maintainers to ask the robot for its "advice" on compensation range. 

- This would ideally be best in a Telegram bot private chat client.  

Let's say that there could be a few algorithmic strategies: 

- ### Target take home pay  
  - This could calculate how much they took home in incentives, and then add their base pay to see the difference. The base pay would be suggested to fill in the gap (positive or negative.) 
  - It would also calculate a new `/multiplier` in order to compensate for the change in "base pay" but still have their take-home be similar. 
  - I think it's attractive for any organization to have the team be mostly on incentives based pay vs "base pay" as this removes stressful management overhead.  
- ### Communication Ability
  - This could assess the communication ability of the contributor by going through the conversations they are apart of and run some sentiment analysis. Examples: 
    - if they receive compliments that's a good thing.
    - if collaborators don't understand what they are saying (unclear communication) that's a bad thing.
- ### Code Quality
  - I decided to make this one specific to code because we can extrapolate this example for any type of deliverable. But, code is the most common deliverable format in the DevPool.
    - We can use ChatGPT to gauge the code quality of pull requests. I think it's pretty straightforward since they have a new product that specializes in this: "code interpreter".

**Comment by Hodlatoor** [comment]
@pavlovcik @sergfeldman 
This is interesting. I think it would be great to see on a UI/UX where you can see median pricing for a bounty for example and then see the applicants by weight based on a system derived qualitative value for the task based on their past experience and contributions. 

It would also be good in a visual format if you are a project manager and you have multiple bounties to flip through and oversee.
